### DISCUSSION TITLE
"like" doesn't work

### DISCUSSION THREAD

**Discussion by lipan1024** [Budibase/budibase#2778] [Help] [answered]
"Like" doesn't work when I use "tables" as a filter, but "like" works when I use "queries" as a filter

**Comment by mjashanks** [comment, marked as answer]
Hi @lipan1024 - can you be more specific? I don't know what yo mean by "Like". Can explain what you are trying to do, maybe with a few screenshots?

**Reply by lipan1024** [reply]
~When I use a table to provide data:
![image](https://user-images.githubusercontent.com/86212306/135233587-f0887c27-df20-4909-b782-1d0371e7ccd7.png)

![image](https://user-images.githubusercontent.com/86212306/135233724-f069a817-e52f-4edc-b929-eca792c8ec0a.png)

![image](https://user-images.githubusercontent.com/86212306/135233834-3f52c183-2068-48b7-b9ac-f90d1eb8c43e.png)
The filter did not work.

~When I use custom query to provide data:
![image](https://user-images.githubusercontent.com/86212306/135234536-9a57f4d5-6b79-4b04-944c-61272473200e.png)

![image](https://user-images.githubusercontent.com/86212306/135234703-3138e84d-0dbb-4cce-b3ee-65ddd64cf627.png)

![image](https://user-images.githubusercontent.com/86212306/135234799-6950ec10-63dd-48a2-a5ba-27f4d9e39aea.png)

This is what I wanted!

**Reply by lipan1024** [reply]
I don't want to use equal, because I want to have data when the filter is empty...

**Reply by lipan1024** [reply]
@mjashanks

**Reply by mjashanks** [reply]
Thanks for the extra info. It seems as though "Like" is broken, as you point out. Also, I think that when your options picker has no value, the condition should not be applied - as you have shown, this is also not the case!

However, you can still achieve what you need, using "Conditions":

1. First of all, add an "All" option to your picker. This will be used to display all values
2. In your Data Provider's filters, remove any filters
3. Now, on your data provider, click "Configure Conditions"
4. Set a condition to "Update Setting" > "Filters", then click "Define Filters"
5. Add your filter, as before, then click save
6. Now, set your condition to only apply if your options picker `Not Equals` > `All`

https://user-images.githubusercontent.com/3524181/135247321-4cd35b40-72b9-45c7-aad2-f5efe63d2783.mp4

I think it would be nice to add a custom `WHERE` clause in here - it would provide alot more flexibility.

**Reply by lipan1024** [reply]
That's a good idea, thank you~

**Comment by v3ss0n** [comment]
Like is still broken it seems.

**Reply by ecxs** [reply]
Like does not seem to work for me either ... version 2.1.28
![Screenshot 2022-11-22 at 07 41 39](https://user-images.githubusercontent.com/4005405/203243101-4f46dbd6-9844-4010-b187-d046e5a72000.png)

**Reply by v3ss0n** [reply]
i think no issue about this is created yet
